src/printer/embed.ts

```diff
@@ -19,135 +19,146 @@ const {
+				...opts,
+				parser: 'typescript',
+			});
 
-	// Frontmatter
-	if (node.type === 'frontmatter') {
-		const frontmatterContent = wrapParserTryCatch(textToDoc, node.value, {
-			...opts,
-			parser: 'typescript',
-		});
+			return [group(['---', hardline, frontmatterContent, hardline, '---', hardline]), hardline];
+		}
 
-		return [group(['---', hardline, frontmatterContent, '---', hardline]), hardline];
-	}
+		// Script tags
+		if (node.type === 'element' && node.name === 'script') {
+			const scriptContent = printRaw(node);
+			const formattedScript = await wrapParserTryCatch(textToDoc, scriptContent, {
+				...opts,
+				parser: 'typescript',
+			});
 
-	// Script tags
-	if (node.type === 'element' && node.name === 'script') {
-		const scriptContent = printRaw(node);
-		let formattedScript = wrapParserTryCatch(textToDoc, scriptContent, {
-			...opts,
-			parser: 'typescript',
-		});
+			// formattedScript = stripTrailingHardline(formattedScript);
+			const isEmpty = /^\s*$/.test(scriptContent);
 
-		formattedScript = stripTrailingHardline(formattedScript);
-		const isEmpty = /^\s*$/.test(scriptContent);
-
-		// print
-		const attributes = path.map(print, 'attributes');
-		const openingTag = group(['<script', indent(group(attributes)), softline, '>']);
-		return [
-			openingTag,
-			indent([isEmpty ? '' : hardline, formattedScript]),
-			isEmpty ? '' : hardline,
-			'</script>',
-		];
-	}
+			// print
+			const attributes = path.map(print, 'attributes');
+			const openingTag = group(['<script', indent(group(attributes)), softline, '>']);
+			return [
+				openingTag,
+				indent([isEmpty ? '' : hardline, formattedScript]),
+				isEmpty ? '' : hardline,
+				'</script>',
+			];
+		}
 
-	// Style tags
-	if (node.type === 'element' && node.name === 'style') {
-		const content = printRaw(node);
-		const supportedStyleLangValues = ['css', 'scss', 'sass'];
-		let parserLang: supportedStyleLang = 'css';
-
-		if (node.attributes) {
-			const langAttribute = node.attributes.filter((x) => x.name === 'lang');
-			if (langAttribute.length) {
-				const styleLang = langAttribute[0].value.toLowerCase();
-				if (supportedStyleLangValues.includes(styleLang))
-					parserLang = styleLang as supportedStyleLang;
+		// Style tags
+		if (node.type === 'element' && node.name === 'style') {
+			const content = printRaw(node);
+			const supportedStyleLangValues = ['css', 'scss', 'sass'];
+			let parserLang: supportedStyleLang = 'css';
+
+			if (node.attributes) {
+				const langAttribute = node.attributes.filter((x) => x.name === 'lang');
+				if (langAttribute.length) {
+					const styleLang = langAttribute[0].value.toLowerCase();
+					if (supportedStyleLangValues.includes(styleLang))
+						parserLang = styleLang as supportedStyleLang;
+				}
 			}
-		}
 
-		return embedStyle(parserLang, content, path, print, textToDoc, opts);
-	}
+			return await embedStyle(parserLang, content, path, print, textToDoc, opts);
+		}
 
-	return null;
+		return;
```

Comment:
what does returning nothing convey? there was nothing to print?

Reply:
Mostly to make typescript happy with the new Prettier v3 definitions.

Reply:
`undefined` or `null` signal Prettier to fallback to "general" print.